src/Pim/Bundle/CatalogBundle/tests/integration/ElasticSearch/IndexConfiguration/PimCatalogNumberIntegration.php

```diff
@@ -0,0 +1,265 @@
+
+    public function testNotEmptyOperator()
+    {
+        $query = $this->createSearchQuery([
+            'query' => [
+                'bool' => [
+                    'filter' => [
+                        'exists' => [
+                            'field' => 'box_quantity-decimal'
+                        ]
+                    ]
+                ]
+            ]
+        ]);
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts(
+            $productsFound,
+            ['product_1', 'product_2', 'product_3', 'product_4', 'product_5', 'product_6']
+        );
+    }
+
+    public function testSortAscending()
+    {
+        $query = $this->createSearchQuery([
+            'query' => [
+                'match_all' => new \stdClass(),
+            ],
+            'sort' => [
+                [
+                    'box_quantity-decimal' => [
+                        'order'   => 'asc',
+                        'missing' => '_first',
+                    ],
+                ],
+            ]
+        ]);
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts(
+            $productsFound,
+            ['product_7', 'product_2', 'product_5', 'product_6', 'product_1', 'product_4', 'product_3']
+        );
+    }
+
+    public function testSortDescending()
+    {
+        $query = $this->createSearchQuery([
+            'query' => [
+                'match_all' => new \stdClass(),
+            ],
+            'sort' => [
+                [
+                    'box_quantity-decimal' => [
+                        'order'   => 'desc',
+                        'missing' => '_last',
+                    ],
+                ],
+            ]
+        ]);
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts(
+            $productsFound,
+            ['product_3', 'product_4', 'product_1', 'product_6', 'product_5', 'product_2', 'product_7']
+        );
+    }
+
+    /**
+     * This method indexes dummy products in elastic search.
+     */
+    protected function addProducts()
+    {
+        $products = [
+            [
+                'sku-varchar'          => 'product_1',
+                'box_quantity-decimal' => '10.0',
+            ],
+            [
+                'sku-varchar'          => 'product_2',
+                'box_quantity-decimal' => '1',
+            ],
+            [
+                'sku-varchar'          => 'product_3',
+                'box_quantity-decimal' => '100.666',
+            ],
+            [
+                'sku-varchar'          => 'product_4',
+                'box_quantity-decimal' => '25.89',
+            ],
+            [
+                'sku-varchar'          => 'product_5',
+                'box_quantity-decimal' => '3.9000',
+            ],
+            [
+                'sku-varchar'          => 'product_6',
+                'box_quantity-decimal' => '7',
+            ],
```

Comment:
Yep, it works! We can do it! we had a first version with alternated quotes and no quotes. Before.